### PR TITLE
[PPP-5166] - Use of Vulnerable Component: Components/jackson-databind 2.14.2 (Non OSGi upgrade)

### PR DESCRIPTION
**Pull request by soagarwal1**

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>



---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 30m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl \
common-dependencies-V1,common-fragment-V1,shims/apache/driver,shims/apache/kar
```
---
:ok_hand: All tests passed!
 
Tests run: **782**, Failures: **0**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/big-data-hadoop-shims/job/PR-1456/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

File: common-dependencies-V1/pom.xml [hunk 1]
```diff
@@ -150,7 +150,7 @@
                 *:hbase-server,*:protobuf-java
               </include>
               <exclude>
-                *:*log4j*,*:xml-apis,*:xercesImpl,*:hadoop*,*:hbase*,*:zookeeper*
+                *:*log4j*,*:xml-apis,*:xercesImpl,*:hadoop*,*:hbase*,*:zookeeper*,com.fasterxml.jackson.core:jackson-databind
```
Comment (NJtwentyone) [inline common-dependencies-V1/pom.xml:153]: Do you what dependency is introducing "com.fasterxml.jackson.core:jackson-databind" ?
are there any other jackson dependencies that are now being introduced?

looking at the artifact pentaho-hadoop-shims-common-dependencies-V1-10.2.0.0-217.jar it contains:

- META-INF/
- META-INF/MANIFEST.MF
- META-INF/maven/
- META-INF/maven/org.pentaho.hadoop.shims/
- META-INF/maven/org.pentaho.hadoop.shims/pentaho-hadoop-shims-common-dependencies-V1/
- META-INF/maven/org.pentaho.hadoop.shims/pentaho-hadoop-shims-common-dependencies-V1/pom.properties
- META-INF/maven/org.pentaho.hadoop.shims/pentaho-hadoop-shims-common-dependencies-V1/pom.xml
- htrace-core-3.1.0-incubating.jar
- httpclient-4.5.14.jar
- mxparser-1.2.2.jar
- pentaho-hadoop-shims-common-mapreduce-10.2.0.0-217.jar
- protobuf-java-2.5.0.jar
- xmlpull-1.1.3.1.jar
- xstream-1.4.20.jar

**Review by NJtwentyone — APPROVED** [review]

File: common-dependencies-V1/pom.xml (hunk 1 above)
Comment (soagarwal1) [reply common-dependencies-V1/pom.xml:153]: Dont remember right now but there were a lot. Mostly the hadoop libraries.

Comment (NJtwentyone) [reply common-dependencies-V1/pom.xml:153]: looking at s/data-integration/system/karaf/system/com/pentaho/hadoop/shims/pentaho-hadoop-shims-ee-apache-driver/10.3.0.0-45/pentaho-hadoop-shims-ee-apache-driver-10.3.0.0-45.jar


the following files were present:

- jackson-annotations-2.14.2.jar
jackson-core-asl-1.9.13.jar
jackson-databind-2.14.2.jar
jackson-jaxrs-1.9.13.jar
jackson-jaxrs-base-2.14.2.jar
jackson-jaxrs-json-provider-2.14.2.jar
jackson-mapper-asl-1.9.13.jar
jackson-module-jaxb-annotations-2.14.2.jar

I'd check a more recent 10.3 build and be parity with the jars and of course update the version.

**Review by NJtwentyone — CHANGES_REQUESTED** [review]
see comment - https://github.com/pentaho/pentaho-hadoop-shims/pull/1456/files#r1765252641

File: common-dependencies-V1/pom.xml (hunk 1 above)
Comment (NJtwentyone) [reply common-dependencies-V1/pom.xml:153]: I re-read the scope, you are suppose to remove.

**Review by NJtwentyone — APPROVED** [review]

**Pull request merged**